docs/topics/spider-middleware.rst

```diff
@@ -122,7 +122,7 @@ object gives you access, for example, to the :ref:`settings <topics-settings>`.
         method (from a previous spider middleware) raises an exception.
 
         :meth:`process_spider_exception` should return either ``None`` or an
-        iterable of :class:`~scrapy.Request` objects and :ref:`item object
+        iterable of :class:`~scrapy.Request` objects or :ref:`item object
```

Comment:
I think this may actually make the sentence more misleading.

With “or”, you could interpret that the iterable must contain either all requests or all items, when it can actually combination of the two.

What’s exactly the potential misinterpretation from the original sentence that you are trying to clarify here? I wonder if switching “item object” to “item object**s**” would fix it.

The point in the sentence is that each object in the iterable may be a Request object or an item object.

Reply:
I think using `and` could be misleading here. Also, the documentation is not consistent with the middleware template file in [middlewares.py.tmpl](https://github.com/scrapy/scrapy/blob/8284de5e7613c47e69d40d4f6a2a1dc846b50dd6/scrapy/templates/project/module/middlewares.py.tmpl#L43): 

```Should return either None or an iterable of Request or item objects.```

I will also change “item object” to “item objects,” as you suggested.

Reply:
I think in the template file things are more clear thanks to the word “objects” being there only at the end. What about doing the same in the documentation?